### ISSUE TITLE
Use fuzzy string matching to link queries to previous queries to suggest the best LLM to respond to user queries

### ISSUE THREAD

**Issue by dctcd** [Antyk45/IBM-Chatbot#95]
Create a function that takes in a String as an input, and inside, it use Fuzzy String Matching to match the queries to all past queries in the database. 

Then get the most similar string found in the database (ensure that similarity is above ~70, else abandon), then get which LLM performed best.

If no rankings are given for any LLMs for the closest query, go to the next closest string. (Although these queries should never be added to the database if no ranking has been given for any LLM)

Ensure that the similarity ranking is still high enough to even consider the string. Repeat until a suitable string is found, if necessary.

Return the LLM with the best performance. If two (or more) LLMs are both ranked the same and are positively ranked (I.e. 4,5 stars each), return both. If one or two are ranked negatively, and the other(s) have no ranking(s), return the non-ranked LLM(s).

In short, return the most positively ranked LLMs, if any, else return the only non-negatively ranked LLMs for the most similar query (above 70 similarity ranking).

Return the LLMs as a list, i.e. if ChatGPT is the best performing LLM for the query, return [“ChatGPT”] else if both AI21 and ChatGPT are both ranked 5 stars for the most similar query, return [“AI21”, “ChatGPT”]
